### PR TITLE
Search and debounce implementation in explore page

### PR DESCRIPTION
**Pull request by danny237**

### CODE REVIEW

File: src/Pages/Explore.jsx [hunk 1]
```diff
@@ -87,6 +132,10 @@ class Explore extends React.Component {
           <div className='search-section'>
             <div className='search-section-input'>
               <InputField
+                value={this.state.search}
+                onChange={(e) => {
+                  this.searchHandler(e.target.value);
+                }}
```
Comment (silwalanish) [inline src/Pages/Explore.jsx:138]:
```suggestion
                onChange={this.searchHandler}
```

File: src/Pages/Explore.jsx [hunk 2]
```diff
@@ -17,41 +19,84 @@ class Explore extends React.Component {
...
 
+  componentDidMount() {
+    this.fetchReconstructions();
+  }
+
+  delayHandler() {
+    this.fetchReconstructions();
+  }
+
+  searchHandler(value) {
```
Comment (silwalanish) [inline src/Pages/Explore.jsx:79]: Take event as param.
```suggestion
  searchHandler(e) {
```

File: src/Pages/Explore.jsx [hunk 3]
```diff
@@ -17,41 +19,84 @@ class Explore extends React.Component {
...
+    this.setState(
+      {
+        search: value,
```
Comment (silwalanish) [inline src/Pages/Explore.jsx:82]:
```suggestion
        search: e.target.value,
```

File: src/Pages/Explore.jsx [hunk 4]
```diff
@@ -17,41 +19,84 @@ class Explore extends React.Component {
...
+        total: 0,
```
Comment (silwalanish) [inline src/Pages/Explore.jsx:83]: Reset page number.
```suggestion
        total: 0,
        page: 1
```

File: src/Pages/Explore.jsx [hunk 5]
```diff
@@ -109,6 +158,15 @@ class Explore extends React.Component {
...
           </div>
+          {this.state.total > 1 && (
+            <Pagination
+              disablePrevious={!this.state.hasPrevious}
+              disableNext={!this.state.hasNext}
+              total={this.state.total}
+              pageChangeHandler={this.pageChangeHandler}
+              page={this.state.page}
+            />
+          )}
```
Comment (silwalanish) [inline src/Pages/Explore.jsx:169]: Center align the pagination.

**Review by silwalanish — APPROVED** [review]

File: src/Pages/Explore.jsx [hunk 6]
```diff
@@ -17,41 +19,84 @@ class Explore extends React.Component {
...
+  }
+
+  delayHandler() {
+    this.fetchReconstructions();
+  }
+
+  searchHandler(e) {
+    this.setState(
+      {
+        search: e.target.value,
```
Comment (silwalanish) [inline src/Pages/Explore.jsx:81]: Let's trim the search query.
```suggestion
        search: e.target.value.trim(),
```

Comment (danny237) [reply src/Pages/Explore.jsx:81]: Ok !

**Review by silwalanish — APPROVED** [review]

**Pull request merged**